presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,546 @@ Tuning Presto
+ * **Description:** Enabling this property forces repartition of joined tables on their keys. This causes an increase in processing time, but reduces the amount of memory required, and therefore allows much larger joins to be performed. After filtering, the index of the table on the left side of the join must fit in memory on each node. This can also be specified on a per-query basis using the ``distributed_index_join`` session property.
+
+
+``distributed-joins-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Use hash distributed joins instead of broadcast joins. Distributed joins require redistributing both tables using a hash of the join key. This can be slower (sometimes substantially) than broadcast joins but allows much larger joins. Broadcast joins require that the tables on the right side of the join after filtering fit in memory on each node whereas distributed joins only need to fit in distributed memory across all nodes. This can also be specified on a per-query basis using the ``distributed_join`` session property.
+
+
+``redistribute-writes``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Force parallel distributed writes. Serves as default value for ``redistribute_writes`` session property. Setting this property will cause write operator to be distributed between nodes. This allows to utilize distributed storage backend especially in case of small number of huge queries.
+
+
+``resources.reserved-system-memory``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``JVM max memory`` * ``0.4``
+ * **Description:** Amount of memory set up as allocation limit for single presto node. Reaching this limit will cause the server to drop operations. Higher value may increase server stability but may cause problems if physical server is used for other purposes as well. In some configurations of host it may even cause presto to be shut down by host if too much memory will be allocated.
+
+
+``sink.max-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``32 MB``
+ * **Description:** Buffer size for IO writes while collecting pipeline results. Higher value may increase speed of IO operations with the cost of additional memory. Also higher value may increase number of data lost when presto node will fail effectively slowing down IO in unstable environment.
+
+
+.. _tuning-pref-exchange:
+
+Exchange properties
+-------------------
+
+``exchange.client-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``25``
+ * **Description:** Number of threads that exchange server can spawn to handle clients. Higher value will increase concurrency but may cause general drop in performance if the value is to high due to context switches and additional memory usage.
+
```

Comment:
Number of threads that exchange server can spawn to handle clients. Higher value will increase concurrency but excessively high values may cause a drop in performance due to context switches and additional memory usage.